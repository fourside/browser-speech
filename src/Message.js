import React from 'react';

export default class Message extends React.Component {
  render() {
    const message = this.props.message;
    return (
      <div className="form-group">
        <label htmlFor="message"> Message </label>
          <textarea id="message" className="form-control" placeholder="type what you want to let me say" 
            value={message} onChange={(e) => this.props.onMessageChange(e)}
          />
      </div>
    );
  }
}

