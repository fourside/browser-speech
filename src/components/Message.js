// @flow
import React from 'react';

type Props = {
  message: string,
  onMessageChange: (x: SyntheticEvent<HTMLTextAreaElement>) => void,
};

export default class Message extends React.Component<Props> {
  messageArea: ?HTMLTextAreaElement;

  componentDidMount() {
    if (this.messageArea) {
      this.messageArea.focus();
    }
  }

  shouldComponentUpdate(nextProps) {
    return this.props.message !== nextProps.message;
  }

  render() {
    const message = this.props.message;
    return (
      <div className="form-group">
        <label htmlFor="message"> Message </label>
          <textarea id="message" className="form-control" placeholder="type what you want to let me say" 
            value={message} onChange={(e) => this.props.onMessageChange(e)}
            ref={(textarea) => {this.messageArea = textarea;}}
          />
      </div>
    );
  }
}

