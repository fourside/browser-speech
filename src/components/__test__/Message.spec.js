import React from 'react';
import Message from '../Message';
import renderer from 'react-test-renderer';

describe('Message', () => {
  it('should be rendered without crashing.', () => {
    const tree = renderer
      .create(<Message message="test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

