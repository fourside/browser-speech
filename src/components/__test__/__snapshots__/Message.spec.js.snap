// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should be rendered without crashing. 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="message"
  >
     Message 
  </label>
  <textarea
    className="form-control"
    id="message"
    onChange={[Function]}
    placeholder="type what you want to let me say"
    value="test"
  />
</div>
`;
